packages/amazonq/test/e2e_new/amazonq/helpers/mcpHelper.ts

```diff
@@ -0,0 +1,308 @@
+ */
+interface MCPServerConfig {
+    scope?: 'global' | 'workspace'
+    name?: string
+    transport?: number
+    command?: string
+    args?: string[]
+    nameEnvironmentVariable?: string
+    valueEnvironmentVariable?: string
+    timeout?: number
+}
+
+const defaultConfig: MCPServerConfig = {
+    scope: 'global',
+    name: 'aws-documentation',
+    transport: 0,
+    command: 'uvx',
+    args: ['awslabs.aws-documentation-mcp-server@latest'],
+    nameEnvironmentVariable: 'hi',
+    valueEnvironmentVariable: 'hi',
+    timeout: 0,
+}
+
+const formItemsMap = {
+    SCOPE: 0,
+    NAME: 1,
+    TRANSPORT: 2,
+    COMMAND: 3,
+    ARGS: 4,
+    ENV_VARS: 6,
+    TIMEOUT: 9,
+} as const
+
+type McpFormItem = keyof typeof formItemsMap
+
+async function selectScope(container: WebElement, scope: string) {
+    try {
+        const a = await container.findElements(By.css('.mynah-form-input-radio-label.mynah-ui-clickable-item'))
+        if (scope === 'global') {
+            const b = a[0]
+            await b.click()
+        } else {
+            const b = a[1]
+            await b.click()
+        }
+    } catch (e) {
+        console.error('Error selecting the scope:', e)
+        throw e
+    }
+}
+
+async function inputName(container: WebElement, name: string) {
+    try {
+        const input = container.findElement(By.css('.mynah-form-input'))
+        await input.sendKeys(name)
+    } catch (e) {
+        console.error('Error inputing the name:', e)
+        throw e
+    }
+}
+
+async function selectTransport(container: WebElement, transport: number) {
+    try {
+        const selectElement = await container.findElement(By.css('select'))
+        const options = await selectElement.findElements(By.css('option'))
+        const optionIndex = transport
+        await options[optionIndex].click()
+    } catch (e) {
+        console.error('Error selecting the transport:', e)
+        throw e
+    }
+}
+
+async function inputCommand(container: WebElement, command: string) {
+    try {
+        const input = container.findElement(By.css('.mynah-form-input'))
+        await input.sendKeys(command)
+    } catch (e) {
+        console.error('Error inputing the command:', e)
+        throw e
+    }
+}
+
+async function inputArgs(container: WebElement, args: string[]) {
+    try {
+        const input = container.findElement(By.css('.mynah-form-input'))
+        const addButton = container.findElement(
+            By.css(
+                '.mynah-button.mynah-button-secondary.fill-state-always.mynah-form-item-list-row-remove-button.mynah-ui-clickable-item'
+            )
+        )
+        for (let i = 0; i < args.length; i++) {
+            await input.sendKeys(args[i])
+            await addButton.click()
+        }
+    } catch (e) {
+        console.error('Error inputing the arguments:', e)
+        throw e
+    }
+}
+
+async function inputEnvironmentVariables(
+    container: WebElement,
+    nameEnvironmentVariable?: string,
```

Comment:
Could we change the typing here to enforce the constraint that its both or nothing?

For example, you could do something like:
```
function f(environmentVariable?: { name: string, value: string })
```